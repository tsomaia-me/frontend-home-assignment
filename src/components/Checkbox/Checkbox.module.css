.root {
    display: flex;
    flex-direction: row;
    align-items: center;
    padding: 12px 0 8px 0;
}

.root > input {
    margin-right: 4px;
}
