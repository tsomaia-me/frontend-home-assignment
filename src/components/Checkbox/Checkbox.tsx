import React from 'react'
import styles from './Checkbox.module.css'

export interface CheckboxProps extends React.ComponentProps<'input'> {
  label?: React.ReactNode
}

function Checkbox({ label, ...restProps }: CheckboxProps) {
  return (
    <label className={styles.root}>
      <input {...restProps} type="checkbox"/>
      <span>{label}</span>
    </label>
  )
}

export default Checkbox
