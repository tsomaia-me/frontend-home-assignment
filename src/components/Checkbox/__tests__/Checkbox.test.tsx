import React from 'react'
import renderer from 'react-test-renderer'
import Checkbox from '../Checkbox'

test('renders Checkbox correctly', () => {
  const json = renderer.create(<Checkbox/>).toJSON()
  expect(json).toMatchSnapshot()
})
