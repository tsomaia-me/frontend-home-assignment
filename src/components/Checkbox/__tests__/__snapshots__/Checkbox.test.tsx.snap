// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Checkbox correctly 1`] = `
<label
  className="root"
>
  <input
    type="checkbox"
  />
  <span />
</label>
`;
